pkg/proxy/ipvs/meta-proxier.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipvs
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/proxy"
+	utilnet "k8s.io/utils/net"
+)
+
+type MetaProxier struct {
```

Comment:
Is there anything about this that is IPVS-specific?  Or could this be elevated and used for iptables, too?

Reply:
> Is there anything about this that is IPVS-specific?

No.

> Or could this be elevated and used for iptables, too?

It was :smiley:

I just moved it to the `ipvs` package in response to https://github.com/kubernetes/kubernetes/pull/80627#pullrequestreview-277510712.

I really wanted to put the MetaProxier in `proxy/util` but it creates a circular depencency; proxy->proxy/util->proxy for the `proxy.ProxyProvider` type.